src/components/views/elements/AppTile.js

```diff
@@ -100,20 +117,46 @@ export default React.createClass({
         }, "mx_IntegrationsManager");
     },
 
+    /* If user has permission to modify widgets, delete the widget, otherwise revoke access for the widget to load in the user's browser
+    */
     _onDeleteClick: function() {
-        console.log("Delete widget %s", this.props.id);
-        this.setState({deleting: true});
-        MatrixClientPeg.get().sendStateEvent(
-            this.props.room.roomId,
-            'im.vector.modular.widgets',
-            {}, // empty content
-            this.props.id,
-        ).then(() => {
-            console.log('Deleted widget');
-        }, (e) => {
-            console.error('Failed to delete widget', e);
-            this.setState({deleting: false});
-        });
+        if (this._canUserModify()) {
+            console.log("Delete widget %s", this.props.id);
+            this.setState({deleting: true});
+            MatrixClientPeg.get().sendStateEvent(
+                this.props.room.roomId,
+                'im.vector.modular.widgets',
+                {}, // empty content
+                this.props.id,
+            ).then(() => {
+                console.log('Deleted widget');
+            }, (e) => {
+                console.error('Failed to delete widget', e);
+                this.setState({deleting: false});
+            });
+        } else {
+            console.log("Revoke widget permissions - %s", this.props.id);
+            this._revokeWidgetPermission();
+        }
+    },
+
+    _deleteWidgetLabel() {
+        if (this._canUserModify()) {
+            return 'Delete widget';
+        }
+        return 'Revoke widget access';
```

Comment:
Oh you do translate this, but it's just far away from this line. Can you add a comment to say that this is being translated? Thanks.